moveit_planners/pilz_industrial_motion_planner/src/trajectory_generator.cpp

```diff
@@ -203,12 +203,12 @@ void TrajectoryGenerator::checkCartesianGoalConstraint(const moveit_msgs::Constr
     throw PositionOrientationConstraintNameMismatch(os.str());
   }
 
-  if (!robot_model_->getJointModelGroup(group_name)->canSetStateFromIK(pos_constraint.link_name))
-  {
-    std::ostringstream os;
-    os << "No IK solver available for link: \"" << pos_constraint.link_name << "\"";
-    throw NoIKSolverAvailable(os.str());
-  }
+  // if (!robot_model_->getJointModelGroup(group_name)->canSetStateFromIK(pos_constraint.link_name))
+  // {
+  //   std::ostringstream os;
+  //   os << "No IK solver available for link: \"" << pos_constraint.link_name << "\"";
+  //   throw NoIKSolverAvailable(os.str());
+  // }
```

Comment:
If not needed anymore: remove completely. Leave a comment here, why it's not needed anymore.
I guess it is redundant to similar checks elsewhere?

But: Instead of doing the error checking and reporting frequently in downstream trajectory generators, why not do it once here?

Reply:
Will double check that this is redundant. Do you think it'd make sense to move the transformation into a tip goal here too?

Reply:
> Do you think it'd make sense to move the transformation into a tip goal here too?

Scratch that part given the transformation is no longer necessary (see reply from above comment)

@rhaschke What would be the easiest way of turning this check into one that can detect subframes? In other places, I've been using `RobotState::knowsFrameTransform()`. 

The method calls are:

`TrajectoryGenerator::generate() -> TrajectoryGenerator::validateRequest() -> TrajectoryGenerator::checkGoalConstraints() -> TrajectoryGenerator::checkCartesianGoalConstraint()`

The first of those methods has access to a planning scene, though I'm unsure if there's a cleaner way than passing a state through the method calls (especially since the state itself is arbitrary, it's just the access to the subframes we're interested in)

Reply:
From my understanding, the check (for existence of the IK (sub)frame and it's usability for IK) didn't fail.
In contrast to `RobotState::knowsFrameTransform()`, `canSetStateFromIK()` should also verify that the IK frame is rigidly connected to the IK solver's tip frame (so exactly what you need).

Reply:
I do get the same error as before uncommenting this when planning with a subframe:

```
[10:15:49.016004] [/fake/move_group]: No IK solver available for link: "tb_1_1/disc_tip"
[10:15:49.016126] [/fake/move_group]: Planner threw an exception (error code: -31): Could not solve request

[10:15:49.017372] [/fake/motion_server]: Failed to plan sequence motion foo. Error code: NO_IK_SOLUTION. If problem persists, check toolbit config.
```

I think the issue the issue is that `JointModelGroup` has no notion of subframes and only checks for links in `canSetFromIK`, whereas `RobotState` knows about both and `setFromIK` is designed for that possibility.